api/src/main/java/de/valtech/aecu/api/groovy/console/bindings/ContentUpgrade.java

```diff
@@ -48,20 +48,29 @@
     ContentUpgrade forChildResourcesOf(String path);
 
     /**
-     * Loops recursive for all child resources of the given path. The path itself is not included.
-     * 
-     * @param path path
-     * @return upgrade object
-     **/
-    ContentUpgrade forDescendantResourcesOf(String path);
+     * Loops recursive for all child resources of the given path. The path itself can be included or not, depending on the value of includeRootResource.
+     *
+     * @param path
+     * @param includeRootResource
+     * @return
+     */
+    ContentUpgrade forDescendantResourcesOf(String path, boolean includeRootResource);
 
     /**
-     * Filters by properties.
+     * Filters by properties. Can be used also for Multi-value properties.
      * 
      * @param conditionProperties properties to filter
      * @return upgrade object
      **/
-    ContentUpgrade filterByProperties(Map<String, String> conditionProperties);
+    ContentUpgrade filterByProperties(Map<String, Object> conditionProperties);
+
+    /**
+     * Filters by multi-value with the given name containing the given conditionValues
+     * @param name
+     * @param conditionValues
+     * @return
```

Comment:
Please provide full. JavaDoc incl. return. Test with "mvn clean javadoc:javadoc" if all is fine.